ui/apps/platform/src/Containers/NetworkGraph/NetworkGraph.tsx

```diff
@@ -77,19 +77,49 @@ function getUrlParamsForEntity(selectEntity: {
 }
 
 export type NetworkGraphProps = {
-    detailType?: UrlDetailTypeValue;
-    detailId?: string;
     model: Model;
 };
 
 export type TopologyComponentProps = NetworkGraphProps;
 
-const TopologyComponent = ({ detailType, detailId, model }: TopologyComponentProps) => {
+function getNodeEdges(selectedNode) {
+    const egressEdges = selectedNode.getSourceEdges();
+    const ingressEdges = selectedNode.getTargetEdges();
+    return [...egressEdges, ...ingressEdges];
+}
+
+function setVisibleEdges(edges) {
+    edges.forEach((edge) => {
+        edge.setVisible(true);
+    });
+}
+
+const TopologyComponent = ({ model }: TopologyComponentProps) => {
+    const { detailType, detailId } = useParams();
     const selectedEntity = detailId && findEntityById(model, detailId, detailType);
     const history = useHistory();
-
     const controller = useVisualizationController();
 
+    if (detailId) {
+        const selectedNode = controller.getNodeById(detailId);
```

Comment:
I notice that if I click a node, and then while that first note is still selected, I click on another node, the URL updates, and the second nodes shows edges, but the first node's edges also still show. 

Nuanced issue, because we don't want to complete deselect the first node, and cause the sidepanel to close--just to re-open immediately--but we still need to `setVisible(false)` the previous edges.

(I wonder if this would help the memory-leak issue.)